sockeye/inference.py

```diff
@@ -255,13 +261,37 @@ def make_input_from_dict(sentence_id: SentenceId,
         tokens = input_dict[C.JSON_TEXT_KEY]
         tokens = list(utils.get_tokens(tokens))
         factors = input_dict.get(C.JSON_FACTORS_KEY)
+        source_prefix = input_dict.get(C.JSON_SOURCE_PREFIX_KEY)
```

Comment:
You define `source_prefix` in `TranslatorInput` as an optional List of Tokens (Tokens are List[str]). I think you should call `utils.get_tokens()` here as well, like for the tokens. It will also save you a number of `split()` calls below.

Reply:
I changed from `source_prefix` to `source_prefix_tokens` to make it align with the use of `tokens` variable. `get_tokens`() is also called here as you suggested. 
Once I added `get_source_prefix_tokens`() to `TranslatorInput` as property as you suggested, `split`() is avoided to be call several times indeed. 
Meanwhile IMO it is actually more intuitive to use `source_prefix_tokens` as a list[str] (i.e. `Tokens`), which is actually the same to tokens. Please take a look at the new revision and let me know if you still prefer a list of Tokens instead of Tokens.

Reply:
Yes you are right, what I mean was `Tokens` not `List[Tokens`. Thanks for the change!